qml/MainScreen.qml

```diff
@@ -217,6 +227,56 @@ Item {
         defaultValue: "file:///usr/share/asteroid-launcher/applauncher/000-default-horizontal.qml"
     }
 
+    ConfigurationValue {
+        id: watchfaceNightstandSource
+        key: "/desktop/asteroid/nightstand/watchface"
+        defaultValue: "file:///usr/share/asteroid-launcher/watchfaces/005-analog-nordic.qml"
+    }
+
+    ConfigurationValue {
+        id: nightstandBrightness
+        key: "/desktop/asteroid/nightstand/brightness"
+        defaultValue: 30
+    }
+
+    ConfigurationValue {
+        id: nightstandDelay
+        key: "/desktop/asteroid/nightstand/delay"
+        defaultValue: 5
+    }
+
+    ConfigurationValue {
+        id: nightstandEnabled
+        key: "/desktop/asteroid/nightstand/enabled"
+        defaultValue: false
+    }
+
+    Item {
+        id: nightstandMode
+        readonly property bool active: ready || (nightstandEnabled.value && holdoff)
+        readonly property bool ready: nightstandEnabled.value && mceCableState.connected
+        property bool holdoff: false
+        property int oldBrightness: 100
+        onReadyChanged: {
+            if (ready) {
```

Comment:
Shouldn't you cancel `nightstandDelayTimer` here ?

Let's say you plug your watch in but it has a bad contact so it:

- gets a connection (ready transitions to true)
- immediately afterward gets a disconnection (ready transitions to false and triggers nightstandDelayTimer)
- gets a connection (ready transitions back to true)

Now the watch stays in that state but the timer elapses and disabled the timer.

---

I think this is maybe the bug you camouflaged behind complicated boolean expressions but with a simplified expression and a more sane logic here this would all work out ?

Reply:
You're right.  I think the current implementation is much cleaner.